utils/utils.py

```diff
@@ -0,0 +1,16 @@
+import ast
+
+def parse_key_value_list(arg):
+    """Helper function to parse a list of key-value pairs into a dictionary.
+    """
+    d = {}
+    for pair in arg.split(","):
+        key, value = pair.split("=", 1)
+        # Try Python literal parsing (handles ints, floats, booleans, None)
+        # https://docs.python.org/3/library/ast.html#ast.literal_eval
+        try:
+            value = ast.literal_eval(value)
+        except (ValueError, SyntaxError):
+            pass
+        d[key] = value
```

Comment:
`value` could be undefined here. In fact, `value` will still be set to the previous value if there was an exception, so not what we want either. We should set it to `None` in the except block.